contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -380,36 +366,46 @@ abstract contract FunctionsBilling is HasRouter, IFunctionsBilling {
     // (1e18 juels/link) * (wei/gas) / (wei/link) = juels per gas
     uint256 juelsPerGas = (1e18 * tx.gasprice) / uint256(weiPerUnitLink);
     // Gas overhead without callback
-    uint96 gasOverheadJuels = uint96(juelsPerGas * commitment.gasOverhead);
-    uint96 costWithoutFulfillment = gasOverheadJuels + commitment.donFee;
+    uint96 gasOverhead = uint96(
+      juelsPerGas * (commitment.gasOverheadBeforeCallback + commitment.gasOverheadAfterCallback)
+    );
 
     // The Functions Router will perform the callback to the client contract
     IFunctionsRouter router = IFunctionsRouter(address(_getRouter()));
     (uint8 result, uint96 callbackCostJuels) = router.fulfill(
-      requestId,
       response,
       err,
       uint96(juelsPerGas),
-      costWithoutFulfillment,
-      msg.sender
+      gasOverhead + commitment.donFee, // costWithoutFulfillment
+      msg.sender,
+      commitment
     );
 
     // Reimburse the transmitter for the fulfillment gas cost
-    s_withdrawableTokens[msg.sender] = gasOverheadJuels + callbackCostJuels;
+    s_withdrawableTokens[msg.sender] = gasOverhead + callbackCostJuels;
     // Put donFee into the pool of fees, to be split later
     // Saves on storage writes that would otherwise be charged to the user
     s_feePool += commitment.donFee;
 
-    emit BillingEnd(
-      requestId,
-      commitment.subscriptionId,
-      commitment.donFee,
-      gasOverheadJuels + callbackCostJuels,
-      gasOverheadJuels + callbackCostJuels + commitment.donFee + commitment.adminFee,
-      FulfillResult(result)
-    );
+    FulfillResult resultCode = FulfillResult(result);
+    if (resultCode == FulfillResult.USER_SUCCESS || resultCode == FulfillResult.USER_ERROR) {
```

Comment:
This is a significant change that should be considered carefully.
However, I think it makes sense to only emit a BillingEnd event if we have called the user's callback and I do not think this event is used in the core node or for obserability, so it is likely fine.

Reply:
I'm just going to roll this back in favor of another PR

Reply:
I'll leave this comment in case someone else thinks otherwise, but after considering this change for a bit, I think it is fine.